modules/core/src/main/java/org/apache/ignite/internal/processors/query/QueryEngine.java

```diff
@@ -36,4 +37,25 @@
      */
     List<FieldsQueryCursor<List<?>>> query(@Nullable QueryContext ctx, String schemaName, String qry, Object... params)
         throws IgniteSQLException;
+
+    /**
+     * Return SQL running queries.
+     *
+     * @return SQL running queries.
+     */
+    List<RunningQueryInfo> runningQueries();
+
+    /**
+     * Return running queries fragments.
+     *
+     * @return Running queries fragments.
+     */
+    List<RunningFragmentInfo> runningFragments();
+
+    /**
+     * Cancel specified query.
+     *
+     * @param qryId Queries ID to cancel.
+     */
+    void cancelQuery(UUID qryId);
```

Comment:
Seems the return type can be a Future, WDYT?

Reply:
Great! I think the default behavior would be async.

Reply:
Ok, I will create separate ticket